mobile/src/main/java/org/openhab/habdroid/core/connection/ConnectionFactory.java

```diff
@@ -100,14 +116,26 @@ public static void shutdown() {
         sInstance.mUpdateThread.quit();
     }
 
+    public static void blockingWaitForInitialization() {
```

Comment:
in my opinion, waitForInitialization would be enough, as this (from my point of view) already implies a blocking behaviour :)